src/idom/_console/update_html_usages.py

```diff
@@ -0,0 +1,231 @@
+from __future__ import annotations
+
+import ast
+import re
+import sys
+from collections.abc import Sequence
+from dataclasses import dataclass
+from keyword import kwlist
+from pathlib import Path
+from textwrap import indent
+from tokenize import COMMENT as COMMENT_TOKEN
+from tokenize import generate_tokens
+from typing import Iterator
+
+import click
+
+from idom import html
+
+
+CAMEL_CASE_SUB_PATTERN = re.compile(r"(?<!^)(?=[A-Z])")
+
+
+@click.command()
+@click.argument("directories", nargs=-1)
+def update_html_usages(directories: list[str]) -> None:
+    """Rewrite files in the given directories to use the new html element API.
+
+    The old API required users to pass a dictionary of attributes to html element
+    constructor functions. For example:
+
+    >>> html.div({"className": "x"}, "y")
+    {"tagName": "div", "attributes": {"className": "x"}, "children": ["y"]}
+
+    The latest API though allows for attributes to be passed as snake_cased keyword
+    arguments instead. The above example would be rewritten as:
+
+    >>> html.div("y", class_name="x")
+    {"tagName": "div", "attributes": {"class_name": "x"}, "children": ["y"]}
+
+    All snake_case attributes are converted to camelCase by the client where necessary.
+
+    ----- Notes -----
+
+    While this command does it's best to preserve as much of the original code as
+    possible, there are inevitably some limitations in doing this. As a result, we
+    recommend running your code formatter like Black against your code after executing
+    this command.
+
+    Additionally, We are unable to perserve the location of comments that lie within any
+    rewritten code. This command will place the comments in the code it plans to rewrite
+    just above its changes. As such it requires manual intervention to put those
+    comments back in their original location.
+    """
+    if sys.version_info < (3, 9):  # pragma: no cover
+        raise RuntimeError("This command requires Python>=3.9")
+    for d in directories:
+        for file in Path(d).rglob("*.py"):
```

Comment:
Does this successfully get all subdirectories as well?

Reply:
Yes `rglob` as apposed to `glob` is recursive.